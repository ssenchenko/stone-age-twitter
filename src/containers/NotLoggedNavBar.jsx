import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import NavButton from '../components/NavButton';

const NavBarStyled = styled.div`
  display: ${({ display }) => (display)};

  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;

  position: relative;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

const NotLoggedNavBar = ({ visible, onSignUpClick, onLoginClick }) => {
  const display = visible ? 'flex' : 'none';
  return (
    <NavBarStyled display={display}>
      <NavButton name="Login" onClickFunc={onLoginClick} />
      <NavButton name="Sign Up" onClickFunc={onSignUpClick} />
    </NavBarStyled>
  );
};

NotLoggedNavBar.propTypes = {
  visible: PropTypes.bool,
  onSignUpClick: PropTypes.func.isRequired,
  onLoginClick: PropTypes.func.isRequired,
};

NotLoggedNavBar.defaultProps = {
  visible: false,
};

export default NotLoggedNavBar;
