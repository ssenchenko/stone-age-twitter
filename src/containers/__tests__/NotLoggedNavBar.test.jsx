import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import NotLoggedNavBar from '../NotLoggedNavBar';
import DefaultTheme from '../../settings/themes';

const onClick = jest.fn();

describe('NotLoggedNavBar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={DefaultTheme}>
          <NotLoggedNavBar visible onSignUpClick={onClick} onLoginClick={onClick} />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
