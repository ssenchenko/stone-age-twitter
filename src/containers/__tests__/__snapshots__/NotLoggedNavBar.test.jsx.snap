// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotLoggedNavBar renders correctly 1`] = `
.c1 {
  border: none;
  outline: none;
  background: #ffffff;
  padding: 15px 32px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
}

.c1:hover {
  color: lightgreen;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<div
  className="c0"
  display="flex"
>
  <button
    className="c1"
    onClick={[MockFunction]}
  >
    Login
  </button>
  <button
    className="c1"
    onClick={[MockFunction]}
  >
    Sign Up
  </button>
</div>
`;
